src/libPersistence/BlockStorage.cpp

```diff
@@ -175,6 +176,58 @@ bool BlockStorage::PutTxBody(const dev::h256& key,
     return (ret == 0);
 }
 
+string MakeKey(const uint64_t& blockNum, const uint32_t& shardId)
+{
+    unsigned int curr_offset = 0;
+    vector<unsigned char> vec;
+    Serializable::SetNumber<uint64_t>(vec, curr_offset, blockNum,
+                                      sizeof(uint64_t));
+    curr_offset += sizeof(uint64_t);
+    Serializable::SetNumber<uint32_t>(vec, curr_offset, shardId,
+                                      sizeof(uint32_t));
+
+    return DataConversion::Uint8VecToHexStr(vec);
+}
+
+bool BlockStorage::PutMicroBlock(const uint64_t& blockNum,
+                                 const uint32_t& shardId,
+                                 const vector<unsigned char>& body)
+{
+    if (!LOOKUP_NODE_MODE)
+    {
+        LOG_GENERAL(WARNING, "Not Expected to be called from non-lookup node");
+        return false;
+    }
+    string key = MakeKey(blockNum, shardId);
+    int ret = m_microBlockDB->Insert(key, body);
+
+    return (ret == 0);
+}
+
+bool BlockStorage::GetMicroBlock(const uint64_t& blockNum,
+                                 const uint32_t& shardId,
+                                 MicroBlockSharedPtr& microblock)
+{
+
+    if (!LOOKUP_NODE_MODE)
```

Comment:
We can remove this check if we already know where this function would be called.